pkg/gardenlet/controller/managedseed/managedseed_controller.go

```diff
@@ -31,9 +31,30 @@ func (c *Controller) managedSeedAdd(obj interface{}) {
 		return
 	}
 
-	if managedSeed.Generation != managedSeed.Status.ObservedGeneration || managedSeed.DeletionTimestamp != nil {
+	generationChanged := managedSeed.Generation != managedSeed.Status.ObservedGeneration
+	jitterUpdates := (c.config.Controllers.ManagedSeed.JitterUpdates != nil) && *(c.config.Controllers.ManagedSeed.JitterUpdates)
+	if managedSeed.DeletionTimestamp != nil {
 		c.logger.Debugf("Added ManagedSeed %s without delay to the queue", key)
 		c.managedSeedQueue.Add(key)
+		return
+	}
+
+	// Newly created managed seed will be enqueued immediatly. Generation is 1 for newly created objects.
+	if managedSeed.Generation == 1 {
+		c.logger.Debugf("Added ManagedSeed %s without delay to the queue", key)
+		c.managedSeedQueue.Add(key)
+		return
+	}
+
+	if generationChanged {
```

Comment:
@ary1992 Not sure but this can be the reason why your tests are failing. Can you check, how does `observed generation` and generation of an object change when we wake up the hibernated shoot?